starter_code/src/movies.js

```diff
@@ -1,20 +1,132 @@
 /* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
 
+function turnHoursToMinutes(movies) {
+    return movies.map(function(movie) {
+        movie = Object.assign({}, movie);
+        var duration = movie.duration.split(' ').reduce(function(acc, time) {
+          if (time.indexOf('h') !== -1) {
+              return Number(time.replace('h', '')) * 60 + acc;
+          } else {
+              return Number(time.replace('min', '')) + acc;
+          }
+        }, 0);
+        movie.duration = duration;
+        return movie;
+    });
+ }
 
 // Get the average of all rates with 2 decimals 
 
+function ratesAverage(arr) {
+    var rates = arr.map(function(movies){
+      // Mira aquí!
+      return Number(movies.rate);
+    });
+    var numbers = rates.map(function(rate){
+      return parseFloat(rate);
+    });
+    var sum = numbers.reduce(function(acc, curr){
+      return acc + curr;
+    }, 0);
+    return sum / numbers.length;
+}
+
 
 // Get the average of Drama Movies
 
+function dramaMoviesRate(films){
+  var justDramas = films.filter(function(movie){
+    if(movie.genre.indexOf('Drama') !== -1) {
+      var dramaRate = movie.rate;
+      if(movie.rate === ''){
+        movie.rate = 0;
+        return movie;
+      }
+      else {
+        return dramaRate;
+      }
+    }
+    else {
+      return 0;
+    }
+  });
+  console.log(justDramas);
+  if(justDramas.length === 0){
+    return undefined;
+  }else{
+    return parseFloat(ratesAverage(justDramas).toFixed(2));
+  }
+}
+
 
 // Order by time duration, in growing order
 
+function orderByDuration(films) { 
+   
+  var sortedFilms = films.map(function(films) {
+    return films;
+   }).sort(function(first, second){
+     if(first.duration === second.duration) {
+      if(first.title > second.title){ //Repasar esto: se puede mejorar
+        return 1;
+      }
+      else if(first.title < second.title){
+        return -1;
+      }
+      else {
+        return 0;
+      }
+     }
+     else {
+       return first.duration - second.duration;
+     }
+  });
+  return sortedFilms;
```

Comment:
Os digo un poco lo mismo, demasiadas condiciones. Además el return 0 os lo podéis ahorrar, ya que si ambas son iguales, os da un poco igual cuál vaya antes.